src/pybind/mgr/cephadm/serve.py

```diff
@@ -709,7 +710,7 @@ def _update_rgw_endpoints(self, rgw_spec: RGWSpec) -> None:
         else:
             self.mgr.remove_health_warning('CEPHADM_RGW')
 
-    def _apply_service(self, spec: ServiceSpec) -> bool:
+    def _apply_service(self, spec: ServiceSpec) -> Tuple[List[DaemonPlacement], List[DaemonPlacement], List[orchestrator.DaemonDescription]]:
```

Comment:
(Advice only) Let's not be afraid to define new types - either by aliasing or by creating a class or named tuple to encapsulate more complex (and long) types.